torch/_dynamo/device_interface.py

```diff
@@ -220,6 +242,29 @@ def get_compute_capability(device: _device_t = None):
         else:
             return torch.cuda.get_device_properties(device).gcnArchName.split(":", 1)[0]
 
+    @staticmethod
+    def is_triton_capable(device: _device_t = None) -> bool:
+        return (
+            torch.version.hip is not None
+            or torch.cuda.get_device_properties(device).major >= 7
+        )
+
+    @staticmethod
+    def raise_if_triton_unavailable(device: _device_t = None) -> None:
```

Comment:
It is kind of wired to throw an exception for this check. Meanwhile, this API has covered `is_triton_capable`. Where does the `is_triton_capable` need to be invoked?

Reply:
This was inspired by how `verify_dynamo` worked, with the original functions `check_if_inductor_supported` and `check_if_dynamo_supported` (which I've now renamed in this PR). An alternative API would be to return a `str | None`, but that means `None` would be the happy (as in, Triton is supported) case, which seems a bit odd.

Implementations of `raise_if_triton_unavailable` should call `is_triton_capable` – that method is only to check whether a device would actually be supported by Triton (as in CUDA below, CUDA on Triton requires a certain device architecture version), not whether we can actually use Triton on the device because of the current operating environment.